content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,857 @@
+
+1. Open `VegetationBending_Common.azsli`.
+
+1. Above your `ProcessBending` function, add a function that calculates the amount of movement and returns the resulting position for a vertex.
+   
+   ```glsl
+   float3 DetailBending(float3 objectSpacePosition, float3 normal, float4 detailBendingParams, float currentTime, float4 worldPosition, float bendLength)
+   {
+      // The information from the vertex colors about how to bend this vertex.
+      float edgeInfo = detailBendingParams.x;
+      float branchPhase = detailBendingParams.y;
+      float branchBendAmount = detailBendingParams.z;
+
+      // Phases (object, vertex, branch)
+      float objPhase = dot(worldPosition.xyz, 2.0); 
+      branchPhase += objPhase;
+      float vtxPhase = dot(objectSpacePosition, branchPhase); 
+
+      // Detail bending for leaves
+      // x: is used for leaves, y is used for branch
+      float2 wavesIn = currentTime;
+      wavesIn += float2(vtxPhase, branchPhase);
+      float4 waves = (frac(wavesIn.xxyy * float4(1.975, 0.793, 0.375,  0.193)) * 2.0 - 1.0) * MaterialSrg::m_detailFrequency * bendLength;
+      waves = abs(frac(waves + 0.5) * 2.0 - 1.0);
+
+      // x: is used for leaves, y is used for branches
+      float2 wavesSum = waves.xz + waves.yw;
+
+      // Leaf and branch bending (xy is used for leaves, z for branches)
+      float3 movement = wavesSum.xxy * float3(edgeInfo * MaterialSrg::m_detailLeafAmplitude * normal.xy, branchBendAmount * MaterialSrg::m_detailBranchAmplitude);
+      return objectSpacePosition + movement;
+   }
+   ```
+
+1. Call the detail bending function in your `ProcessBending` function. Inside the conditional, add the call:
+   
+   ```glsl
+   if (o_color_isBound) 
+   {
+      // Overall wind
+      float4 currentBending = SetUpWindBending(currentTime, worldPosition);
+
+      // Detail bending
+      float3 currentOutPosition = DetailBending(objectSpacePosition, normal, detailBendingParams, currentTime, worldPosition, currentBending.w);
+   }
+
+   return adjustedWorldPosition;
+   ```
+
+   {{< note >}}
+   The `currentBending.w` parameter that's passed into the `DetailBending` function controls the overall bending length according to the wind's strength and direction. 
+   {{< /note >}}
+
+1. You will need to set the actual vertex shader outputs to use the output from the detail bending function. Set the world position so the code following the conditional can update the positions correctly. 
+   
+   ```glsl
+   if (o_color_isBound) 
+   {
+      // Overall wind
+      float4 currentBending = SetUpWindBending(currentTime, worldPosition);
+
+      // Detail bending
+      float3 currentOutPosition = DetailBending(position, normal, detailBendingParams, currentTime, worldPosition, currentBending.w);
+
+      adjustedWorldPosition = mul(objectToWorld, float4(currentOutPosition, 1.0));
+   }
+
+   return adjustedWorldPosition;
+   ```
```

Comment:
Since both of these steps happen inside the conditional, I suggest consolidating them and having only one code snippet. I also tried to rephrase the second step - please verify if this is still accurate. 

The suggestion below would look like: 
![image](https://user-images.githubusercontent.com/75444793/182721793-c11b4966-63e0-43d3-b9e2-2ab9deb2d612.png)

````suggestion
1. In the `ProcessBending` function, inside the conditional:

   * Call the `DetailBending` function.
   
   * Set and return the adjusted world position, so the actual vertex shader output uses the output from the `DetailBending` function. 
   
   ```hlsl
   if (o_color_isBound) 
   {
      // Overall wind
      float4 currentBending = SetUpWindBending(currentTime, worldPosition);

      // Detail bending
      float3 currentOutPosition = DetailBending(position, normal, detailBendingParams, currentTime, worldPosition, currentBending.w);

      adjustedWorldPosition = mul(objectToWorld, float4(currentOutPosition, 1.0));
   }

   return adjustedWorldPosition;
   ```
   
   {{< note >}}
   The `currentBending.w` parameter that's passed into the `DetailBending` function controls the overall bending length according to the wind's strength and direction. 
   {{< /note >}}
````